### PR TITLE
Add solph mixin classes

### PR DESCRIPTION
**Pull request by simnh**
This is linked to #280 . With this structure we are even able to have multiple Mixin-classes

### CODE REVIEW

**Comment by uvchik** [comment]
I gernerally like the proposal. I think we should keep in mind what @ckaldemeyer said [here](https://github.com/oemof/oemof/issues/280#issuecomment-283274646) but I think in this case we just add new methods. It is a very simple construction and therefore it will be easy to improve or change it later.

1. How is the call? Like this:

```python
om.flow_share(flow_object1, flow_object2, 0.5)
```

2. I would add a typecheck (check if f1 is of type flow) . It makes debugging easier.

3. Is it possible to pass the labels (string) instead of the objects? How do I get the flows? Like this:

```python
flow_object1 = om.flows[energysystem.groups['label_source'], energysystem.groups['label_target']]
# The short version would be nicer in may opinion:
flow_object1 = ('label_source', 'label_target')
```

**Comment by simnh** [comment]
I think this is obsolete for now, see after version 0.2

**Comment by uvchik** [comment]
A pitty, I really liked the idea.

**Comment by simnh** [comment]
yeah we still do not have a solution for Constraints....thats right...

**Comment by ckaldemeyer** [comment]
> yeah we still do not have a solution for Constraints....thats right...

What about just writing a module with functions that receive the `es` or `om` as params and additional params to set constraints!? This would also be straightforward..

**Comment by simnh** [comment]
yes I think we could do that as a start
